### DISCUSSION TITLE
Android App development support

### DISCUSSION THREAD

**Discussion by huutho-dev** [PRProd/HA-Firemote#376] [General]
### Feature Request

I want run all function in Android application

**Comment by PRProd** [comment]
Greetings @huutho-dev!

Could you explain what you are asking for with a little bit more detail?  I don't understand what you want.

**Comment by huutho-dev** [comment]
Hello @PRProd 
I find this repo very high quality, hope I can find something useful here. 
I want to create an Android Application can control (Remote control and Cast or Stream Video) Amazone FireTV / Roku/ Chromecast ...

**Comment by PRProd** [comment]
Since this isn't a bug / issue / Feature Request, I'm converting it to a discussion.
<br>

@huutho-dev,

Thank you for calling my work *very high quality*!  I appreciate the compliment!

I'm willing to help you in any way I can, however I don't know how much I'll be able to contribute since my work is very specifically tailored to be used inside of the [Home Assistant](https://www.home-assistant.io/) ecosystem to command external media devices.

What I mean by that is this: If you take a look at the [Firemote wiki](https://github.com/PRProd/HA-Firemote/wiki), near the top you will find a section titled [How Does Firemote Send commands to my devices?](https://github.com/PRProd/HA-Firemote/wiki#how-does-firemote-send-commands-to-my-devices) where this diagram exists:
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/c5591466-3d3b-4bf3-831c-296060838b95)
The Firemote card (this repo) is only a small portion of what is required to establish communications with external media player devices.

If/when you are in need of ADB commands to communicate with Amazon Fire and/or other Android based media players, I have a wealth of knowledge for you there.  Here are some strong (non exhaustive) examples for you to look at:
 - [ADB Commands for onn. Google TV 4K Streaming Box (2023)](https://github.com/PRProd/HA-Firemote/wiki/onn.-Google-TV-4K-Streaming-Box-(2023)#adb-commands)
 - [ADB Commands for Fire TV Stick 4K 2nd Gen (2023)](https://github.com/PRProd/HA-Firemote/wiki/Fire-TV-Stick-4K-2nd-Gen-(2023)#adb-commands)
 - [ADB commands for Fire TV Stick 4K Max 2nd Gen (2023)](https://github.com/PRProd/HA-Firemote/wiki/Fire-TV-Stick-4K-Max-2nd-Gen-(2023)#adb-commands)
 - [ADB commands for Smart TV Fire TV 4 Series (2021)](https://github.com/PRProd/HA-Firemote/wiki/Smart-TV---Fire-TV-4-Series-(2021)#adb-commands)
 - [ADB and Home Assistant commands that Firemote currently uses for each button](https://github.com/PRProd/HA-Firemote/blob/v3.3.3/dist/HA-Firemote.js#L4681-L5444)
 - [Launch commands and artwork for over 200 apps currently used by Firemote](https://github.com/PRProd/HA-Firemote/blob/v3.3.3/dist/launcher-buttons.js)


I've spent a lot of time developing the [README.md](https://github.com/PRProd/HA-Firemote/blob/main/README.md) file and the [Firemote Wiki](https://github.com/PRProd/HA-Firemote/wiki) entries, so I hope that you can find some helpful resources there as well.  

I'm happy to help where I can!

-Doug
